src/java.base/share/lib/security/default.policy

```diff
@@ -103,6 +103,14 @@ grant codeBase "jrt:/jdk.accessibility" {
     permission java.lang.RuntimePermission "accessClassInPackage.sun.awt";
 };
 
+grant codeBase "jrt:/jdk.attach" {
+    permission java.lang.RuntimePermission "accessClassInPackage.com.ibm.tools.attach.target";
+    permission java.lang.RuntimePermission "accessClassInPackage.com.ibm.oti.util";
+    permission java.util.PropertyPermission "com.ibm.tools.attach.*", "read";
+    permission java.io.FilePermission "<<ALL FILES>>", "read,write,delete";
```

Comment:
Can you add a comment that explains why attach needs read/write/delete for all files?

Reply:
Yeah, added comments about the class/method requiring the permission specified.

In addition, following are the stacktrace when such permissions are missing:
```
Exception in thread "main" java.security.AccessControlException: Access denied ("java.io.FilePermission" "/tmp/.com_ibm_tools_attach" "read")
	at java.base/java.security.AccessController.throwACE(AccessController.java:176)
	at java.base/java.security.AccessController.checkPermissionHelper(AccessController.java:237)
	at java.base/java.security.AccessController.checkPermission(AccessController.java:373)
	at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:322)
	at java.base/java.lang.SecurityManager.checkRead(SecurityManager.java:661)
	at java.base/java.io.File.exists(File.java:815)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9AttachProvider.listVirtualMachinesImp(OpenJ9AttachProvider.java:126)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9AttachProvider.lambda$listVirtualMachines$1(OpenJ9AttachProvider.java:114)
	at com.ibm.tools.attach.attacher.OpenJ9AttachProvider$$Lambda$49.0000000044482AF0.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:647)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9AttachProvider.listVirtualMachines(OpenJ9AttachProvider.java:115)

Exception in thread "main" java.security.AccessControlException: Access denied ("java.io.FilePermission" "/tmp/.com_ibm_tools_attach/62249/replyInfo" "write")
	at java.base/java.security.AccessController.throwACE(AccessController.java:176)
	at java.base/java.security.AccessController.checkPermissionHelper(AccessController.java:237)
	at java.base/java.security.AccessController.checkPermission(AccessController.java:373)
	at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:322)
	at java.base/java.lang.SecurityManager.checkWrite(SecurityManager.java:752)
	at java.base/java.io.RandomAccessFile.<init>(RandomAccessFile.java:247)
	at java.base/java.io.RandomAccessFile.<init>(RandomAccessFile.java:214)
	at java.base/com.ibm.tools.attach.target.Reply.writeReply(Reply.java:79)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.tryAttachTarget(OpenJ9VirtualMachine.java:447)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.attachTargetImpl(OpenJ9VirtualMachine.java:154)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.lambda$attachTarget$1(OpenJ9VirtualMachine.java:129)
	at com.ibm.tools.attach.attacher.OpenJ9VirtualMachine$$Lambda$50.00000000640F0450.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:703)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.attachTarget(OpenJ9VirtualMachine.java:131)


ck 001 get shutdownTimeoutMs
Exception in thread "main" java.security.AccessControlException: Access denied ("java.io.FilePermission" "/tmp/.com_ibm_tools_attach/62249/replyInfo" "delete")
	at java.base/java.security.AccessController.throwACE(AccessController.java:176)
	at java.base/java.security.AccessController.checkPermissionHelper(AccessController.java:237)
	at java.base/java.security.AccessController.checkPermission(AccessController.java:373)
	at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:322)
	at java.base/java.lang.SecurityManager.checkDelete(SecurityManager.java:780)
	at java.base/java.io.File.delete(File.java:1048)
	at java.base/com.ibm.tools.attach.target.Reply.deleteReply(Reply.java:145)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.tryAttachTarget(OpenJ9VirtualMachine.java:511)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.attachTargetImpl(OpenJ9VirtualMachine.java:154)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.lambda$attachTarget$1(OpenJ9VirtualMachine.java:129)
	at com.ibm.tools.attach.attacher.OpenJ9VirtualMachine$$Lambda$50.0000000090101460.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:703)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.attachTarget(OpenJ9VirtualMachine.java:131)
```